test/e2e/test_manufacturer.py

```diff
@@ -140,70 +199,147 @@ def test_update(test_client):
     manufacturer_post = {
         "name": "Manufacturer A",
         "url": "http://example.com/",
-        "address": "Street A",
+        "address": {
+            "building_number": "1",
+            "street_name": "Example Street",
+            "town": "Oxford",
+            "county": "Oxfordshire",
+            "country": "United Kingdom",
+            "postcode": "OX1 2AB",
+        },
+        "telephone": "0932348348",
     }
 
-    response = test_client.post("/v1/manufacturer", json=manufacturer_post)
+    response = test_client.post("/v1/manufacturers", json=manufacturer_post)
 
     manufacturer_patch = {
         "name": "Manufacturer B",
         "url": "http://test.co.uk/",
-        "address": "Street B",
+        "address": {"building_number": "2"},
+        "telephone": "07569585584",
     }
-    response = test_client.patch(f"/v1/manufacturer/{response.json()['id']}", json=manufacturer_patch)
+    response = test_client.patch(f"/v1/manufacturers/{response.json()['id']}", json=manufacturer_patch)
 
     assert response.status_code == 200
     manufacturer = response.json()
 
     assert manufacturer["name"] == manufacturer_patch["name"]
     assert manufacturer["url"] == manufacturer_patch["url"]
-    assert manufacturer["address"] == manufacturer_patch["address"]
+    assert manufacturer["address"]["building_number"] == manufacturer_patch["address"]["building_number"]
+    assert manufacturer["telephone"] == manufacturer_patch["telephone"]
+
+
+def test_partial_address_update(test_client):
+    """Test updating a manufacturer's address"""
+    manufacturer_post = {
+        "name": "Manufacturer A",
+        "url": "http://example.com/",
+        "address": {
+            "building_number": "1",
+            "street_name": "Example Street",
+            "town": "Oxford",
+            "county": "Oxfordshire",
+            "country": "United Kingdom",
+            "postcode": "OX1 2AB",
+        },
+        "telephone": "0932348348",
+    }
+
+    response = test_client.post("/v1/manufacturers", json=manufacturer_post)
+
+    manufacturer_patch = manufacturer_post = {
+        "address": {
+            "street_name": "test",
+        }
+    }
+    response = test_client.patch(f"/v1/manufacturers/{response.json()['id']}", json=manufacturer_patch)
+
+    assert response.status_code == 200
+    manufacturer = response.json()
+
+    assert manufacturer["address"]["street_name"] == manufacturer_patch["address"]["street_name"]
```

Comment:
Should be asserting the whole address to make sure that no unexpected changes were made to the other address values.
```suggestion
    assert manufacturer["address"] == {**manufacturer_post["address"], **manufacturer_patch["address"]}
```